pkg/controller/migmigration/stage.go

```diff
@@ -529,6 +529,64 @@ func (t *Task) stagePodReport(client k8sclient.Client) (report PodStartReport, e
 	return
 }
 
+// Match number of stage pods in source and destination cluster
+func (t *Task) allStagePodsMatch() (report []string, err error) {
+	dClient, err := t.getDestinationClient()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	podDList := corev1.PodList{}
+
+	options := k8sclient.MatchingLabels(t.stagePodLabels())
+	err = dClient.List(context.TODO(), options, &podDList)
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	sClient, err := t.getSourceClient()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	podSList := corev1.PodList{}
+	err = sClient.List(context.TODO(), options, &podSList)
+
+	dPods := make(map[string]string)
+
+	for _, pod := range podDList.Items{
+		dPods[pod.Name] = pod.Name
+	}
+
+	for _, pod := range podSList.Items{
```

Comment:
Going to merge and follow up with some changes here I think we can simplify the for block some.